azure-pipelines.yml

```diff
@@ -106,27 +106,6 @@ jobs:
         LOCK_FILE: './build_tools/azure/python_nogil_lock.txt'
         COVERAGE: 'false'
 
-# Check compilation with intel C++ compiler (ICC)
-- template: build_tools/azure/posix.yml
-  parameters:
-    name: Linux_Nightly_ICC
-    vmImage: ubuntu-20.04
-    dependsOn: [git_commit, linting]
-    condition: |
-      and(
-        succeeded(),
-        not(contains(dependencies['git_commit']['outputs']['commit.message'], '[ci skip]')),
-        or(eq(variables['Build.Reason'], 'Schedule'),
-           contains(dependencies['git_commit']['outputs']['commit.message'], '[icc-build]')
-        )
-      )
-    matrix:
-      pylatest_conda_forge_mkl:
-        DISTRIB: 'conda'
-        LOCK_FILE: 'build_tools/azure/pylatest_conda_forge_mkl_no_coverage_linux-64_conda.lock'
-        COVERAGE: 'false'
-        BUILD_WITH_ICC: 'true'
-
```

Comment:
I think we should keep this, assuming setting the `CC` env variable works for setuptools. But it's true it would be better to test it at some point. But I would not make this a blocker for this PR.

Reply:
I did not get the Intel compilers to work by setting the `CC` environment variable (and friends). From memory, I got Intel to compile by setting environment variables, but there was an issue when running `pytest`.

(The Azure logs does not exist anymore because it has been too long)

Reply:
Ok let's keep that in mind for a follow-up PR.